code/clue.js

```diff
@@ -1,59 +1,238 @@
-// The values will be set later.
+let mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+}
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years young`
+  document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation
+  document.getElementById('killerDescription').innerHTML = mystery.killer.description
+  document.getElementById('killerImage').src = mystery.killer.image
+}
+document.getElementById('killerCard').onclick = pickKiller
+
+// FUNCTIONS pickWeapon 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponName').innerHTML = mystery.weapon.name
+  document.getElementById('weaponFact').innerHTML = `${mystery.weapon.fact}`
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight} lbs`
+
 }
+document.getElementById('weaponCard').onclick = pickWeapon
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//favouriteWeapon
+const favouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+}
+const shuffleFavouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+  document.getElementById('favouriteWeapon').innerHTML
+}
+shuffleFavouriteWeapon()
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+//This block of code is the one I really wanted to make work but somehow it never did:
+
+// const favourite = suspects.favouriteWeapon
+// const shuffleFavouriteWeaponDuplicate = new Array(6).fill(favourite)
+// const randomArray = [...weapons, ...favouriteWeaponDuplicate]
+// const pickNotSoRandomWeapon = () => {
+//   mystery.favouriteWeapon = randomArray[Math.floor(Math.random() * randomArray.length)]
+// }
+// pickNotSoRandomWeapon()
+
+// pickRoom
+const pickRoom = () => {
+  mystery.rooms = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = `${mystery.rooms}`
+}
+document.getElementById('roomCard').onclick = pickRoom
+
+// FUNCTION revealMystery that will be invoked when you click that button
+const thisMystery = document.getElementById('mystery')
```

Comment:
I suggest to group all the declarations together if you were to use them somewhere else, for example in top of the page. Same with the functions and the executions. 😍